### PR TITLE
Display more useful errors in the UI

### LINKED ISSUE #42: API/UI are not providing useful hints when recipe creation fails
What I observed is that when a recipe is created on Zimfarm and it fails, `zimit-frontend` API is always returning a 500 error and no error message is provided to the UI.

Looking at the code, I realize there are 3 operations performed on the Zimfarm when someone requests a youzim.it execution: the schedule creation, the requested task creation and the schedule deletion.

For every of these requests, there are many different situations to take into account when speaking about a failing HTTP request:
1. Zimfarm returns a 400 error with error description, i.e. an explanation about which recipe flag(s) is(are) incorrect
1. Zimfarm returns a 400 error without error description
1. Zimfarm returns a 4xx error
1. Zimfarm returns a 5xx error
1. Zimfarm call fails at transport level (timeout, DNS issue, ...)
1. Zimfarm returns a success code but we do not find what we expect in the response

Currently the code:
- raises an `InternalError` exception when schedule creation fails (no matter the reason)
- does not check the response of schedule creation (not very useful indeed)
- raises an `InternalError` exception when requested task creation fails
- raises a `ValueError` exception when we do not find the expected task ID in the response of requested task creation
- only logs details when the schedule deletion fails and continue with normal operation

`ValueError` exceptions are not handled by web server so they probably lead to 500 error at HTTP level (if I'm not mistaken).

`InternalError` exceptions are handled as 500 error but the exception message is lost.

A minimal intervention would be to return the `InternalError` message in the HTTP response and display it in the UI, however I'm not convinced that it always make sense for all situations. Or I'm even convinced it does not makes sense in many situation. Many error could be pretty obscure to the end user (and this is most probably why it has been done like that so far).

What I would suggest is:
1. when the Zimfarm returns a 400 error with error description, forward the error description in the API response and display this description in the UI
2. when there is a transport level error, return a 503 error and process 503 error at the UI level with a nice message indicating that the service is currently unavailable but the user should try again in few hours
3. replace `ValueError` by `InternalError`, since this does not make a difference (except if I'm mistaken)
4. return `InternalError` message to the UI but do not display it (developers / tech users could more easily identify the issue but it is too complex information to present to the regular user)

### PR DESCRIPTION
**Pull request by benoit74**
## Rationale
- Fix #42 
- automatize deployment to k8s (⚠️ `ZIMIT_KUBE_CONFIG` secret must be set in repo before merging ⚠️)
- align CI with Python bootstrap conventions

## Changes
- add a development stack based on Docker-Compose to test the whole stuff (Zimit + Zimfarm) locally
- simplify the Exception handling logic in Zimit API to produce consistent results / simplify code maintenance
- in case of Zimfarm error, log the Zimfarm response in API under all all log levels (not just DEBUG) since this is useful (including for detecting what is most common user error) and not supposed to happen a lot (UI should prevent user to do silly things)
- change first part of Zimit UI error message from "Unable to create schedule" to "Unable to request ZIM creation". At this stage of the message this is the only thing we know. Details are added on the next line (did we encountered an error at schedule creation or task request).
- simplify second part of Zimit UI error message which was handling many potential situation:
  - stop considering that API response may contain either `error` or `error_description` or `message` fields ; in fact this response comes from Zimit API and we know there is only a potential `error` field
  - if `error` field is present in the API response, display it "as-is" and without Zimit API HTTP response code (this is too technical for the end user + we return almost always the same one) ; display it no matter the HTTP response code (it was only done for HTTP 400 before)
  - otherwise, display API HTTP response status code + status text message (as before)
 

NOTA:
- this still displays the Zimfarm HTTP error code because it is impossible to retrieve this info from the UI otherwise, even for an expert user
- the fact that "INTERNAL SERVER ERROR" is in capital letters is probably only a side-effect link to the fact that in the dev stack I do not use nginx reverse proxy but directly the Flask app

## Screenshots
Before:
image

After, when API provides error data:
image

After, when API does not provide error data but just HTTP status code (and maybe other data we do not know how to use):
image

### CODE REVIEW

**Comment by benoit74** [comment]
NOTA: CI has gone wild, most probably due to my force push, `code-formating` does not exists anymore and has been replaced by `QA/check-qa`

File: .github/workflows/Publish.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        uses: actions/checkout@v3
+
+      - name: Build and publish Docker Image
+        uses: openzim/docker-publish-action@v10
+        with:
+          image-name: openzim/zimit-ui
+          on-master: latest
+          restrict-to: openzim/zimit-frontend
+          registries: ghcr.io
+          credentials: GHCRIO_USERNAME=${{ secrets.GHCR_USERNAME }}
```
Comment (rgaudin) [inline .github/workflows/Publish.yml:25]: Use an empty first line so both lines are aligned. Those things are typo hell so whatever helps avoid them if positive

File: ui/src/constants.js [hunk 2]
```diff
@@ -98,19 +98,16 @@ export default {
...
     }
-    let status_text = response.statusText ? response.statusText : statuses[response.status];
-    if (response.status == 400) {
-      if (response.data && response.data.error)
-        status_text += "<br />" + JSON.stringify(response.data.error);
-      if (response.data && response.data.error_description)
-        status_text += "<br />" + JSON.stringify(response.data.error_description);
-      if (response.data && response.data.message)
-        status_text += "<br />" + JSON.stringify(response.data.message);
+    // If error is provided, display it (do not display error code since this is too technical)
```
Comment (rgaudin) [inline ui/src/constants.js:105]: Beware, you're turning into @Popolechien 😉

**Review by rgaudin — APPROVED** [review]
Great!
Can you align the lines in workflow and merge?

File: ui/src/constants.js (hunk 2 above)
Comment (benoit74) [reply ui/src/constants.js:105]: How dare you? 🤣

**Pull request merged**